### ISSUE TITLE
Citeback references appear on the wrong page

### ISSUE THREAD

**Issue by ewiddison** [BYU-Engineering/thesis_template#9]
Sometimes the "Cited on" text in the references section appears on the page before the reference that it refers to. In the attached document, citation 13 is on page 16 but its "Cited on" is on page 15.
[PageBugFancyTwopageTemplate.zip](https://github.com/BYU-Engineering/thesis_template/files/13311155/PageBugFancyTwopageTemplate.zip)

**Comment by jgoeders** [comment]
<img width="620" alt="image" src="https://github.com/BYU-Engineering/thesis_template/assets/1851142/b1e2c037-2553-479c-99a8-112ff16c0597">

**Comment by jgoeders** [comment]
@ewiddison  I did reproduce your issue (see above), but this appears to be fixed in the latest version.  

<img width="577" alt="image" src="https://github.com/BYU-Engineering/thesis_template/assets/1851142/bbf76691-504d-46e0-bcfd-9d971795b370">

**Comment by hrjs84** [comment]
Unfortunately I am having this same issue, using the most recent version. For me it happens on every page break. I am on Windows, using TeXLive, and the TeXWorkshop extension for VS Code using the workflow I documented in #4 .
![image](https://github.com/BYU-Engineering/thesis_template/assets/44150209/8f7b9f55-a570-420e-8bf2-a69e246c67b0)
I notice in your example that reference 12 has been moved to the next page, perhaps something else changed in that document that shifted it causing it to appear correctly. For me, moving between previous and current versions of the template keeps my references on the same page, and in both versions the citeback is incorrectly placed.
